tests/framework/artifacts.py

```diff
@@ -236,204 +103,24 @@ def snapshot_version(self):
-            ArtifactCollection.ARTIFACTS_KERNELS,
-            ArtifactCollection.MICROVM_KERNEL_EXTENSION,
-            ArtifactType.KERNEL,
-            Artifact,
-            keyword=keyword,
-        )
-
-        supported_kernels = {f"vmlinux-{sup}.bin" for sup in select_supported_kernels()}
-        valid_kernels = [
-            kernel for kernel in kernels if Path(kernel.key).name in supported_kernels
-        ]
-
-        return valid_kernels
-
-    def disks(self, keyword=None):
-        """Return disk artifacts for the current arch."""
-        return self._fetch_artifacts(
-            ArtifactCollection.ARTIFACTS_DISKS,
-            ArtifactCollection.MICROVM_DISK_EXTENSION,
-            ArtifactType.DISK,
-            DiskArtifact,
-            keyword=keyword,
-        )
-
-
-class SnapshotType(Enum):
-    """Supported snapshot types."""
-
-    FULL = 0
-    DIFF = 1
-
-
-class SnapshotMemBackendType(Enum):
-    """
-    Supported guest memory backend types used for snapshot load.
-
-    - `FILE`: establishes if the guest memory is backed by a file.
-    - `UFFD`: indicates that the guest memory page faults are handled by
-              a dedicated UFFD page-fault handler process.
-    """
-
-    FILE = "File"
-    UFFD = "Uffd"
-
-
-class Snapshot:
-    """Manages Firecracker snapshots."""
-
-    def __init__(self, mem, vmstate, disks, net_ifaces, ssh_key):
-        """Initialize mem, vmstate, disks, key."""
-        assert mem is not None
-        assert vmstate is not None
-        assert disks is not None
-        assert ssh_key is not None
-        self._mem = mem
-        self._vmstate = vmstate
-        self._disks = disks
-        self._ssh_key = ssh_key
-        self._net_ifaces = net_ifaces
-
-    def rebase_snapshot(self, base):
-        """Rebases current incremental snapshot onto a specified base layer."""
-        run_rebase_snap_bin(base.mem, self.mem)
-        self._mem = base.mem
-
-    def cleanup(self):
-        """Delete the backing files from disk."""
-        os.remove(self._mem)
-        os.remove(self._vmstate)
-
-    @property
-    def mem(self):
-        """Return the mem file path."""
-        return self._mem
-
-    @property
-    def vmstate(self):
-        """Return the vmstate file path."""
-        return self._vmstate
-
-    @property
-    def disks(self):
-        """Return the disk file paths."""
-        return self._disks
-
-    @property
-    def ssh_key(self):
-        """Return the ssh key file path."""
-        return self._ssh_key
-
-    @property
-    def net_ifaces(self):
-        """Return the list of net interface configs."""
-        return self._net_ifaces
+def firecracker_artifacts():
+    """Return all supported firecracker binaries."""
+    cargo_version = get_firecracker_version_from_toml()
+    # until the next minor version (but *not* including)
+    max_version = (cargo_version.major, cargo_version.minor + 1, 0)
+    min_version = (1, 2, 0)
```

Comment:
nit: Can we turn our relate policy into code instead of hardcoding 1.2? :thinking: